### PR TITLE
fix(intelligent-assistant): refactored mcp config modal

### PR DESCRIPTION
**Pull request by ciiay**
## Hey, I just made a Pull Request!

For [RHIDP-14843](https://redhat.atlassian.net/browse/RHIDP-14843)

Refactor the MCP server configure modal with PatternFly layout and improved credential management:

- Add **Use organization default token** / **Use personal token** radio selectors when an app-config default token is available, replacing the remove-personal-token flow
- Show server status, tools list, and enabled toggle with local draft state applied on Save
- Hide the tools section when the server is disabled; keep status consistent while drafting an unverified personal token
- Disable Save after a failed token validation until the input changes
- Auto-enable the server after a successful personal token save
- Add sortable **Name** and **Status** columns to the MCP servers table, with always-visible sort icons (gray when inactive, directional when active)
- Keep the DCR configure modal aligned with personal-token servers (status, tools, enabled toggle, and Save/Cancel) with an additional DCR info alert and no token input
- Show **Fetching status...** in the Status section while server tools/status are loading
- Bump `@patternfly/react-core` to 6.6.0 and add translations for new modal strings (de, es, fr, it, ja)

The backend now exposes `hasOrgToken` on MCP server list and patch responses so the UI can distinguish organization default tokens from personal-only servers.

Behavior confirmation: ~~[Intelligent-assistant MCP config modal details alignment](https://docs.google.com/document/d/19LSi9GWJ6GmkKWKvBiD6HgDDev7Ik-dbzGXlPsWlfpM/edit?usp=sharing)~~

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/redhat-developer/rhdh-plugins/blob/main/CONTRIBUTING.md#creating-changesets))
- [ ] Added or Updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

#### Test steps:
[Test steps for Intelligent-assistant plugin](https://docs.google.com/document/d/1RfFFSgLhih1uXnxmwb2r0UUHlR1lMgrlxCZuDRKN_48/edit?usp=sharing)

screen recording(updated on 7/13)

https://github.com/user-attachments/assets/7d87d639-1f4d-447b-a582-7be2942dae7f

### CODE REVIEW

**Review by HusneShabbir — DISMISSED** [review]
Hey @ciiay , I have a couple of questions:

1. Why haven't we implemented the **"forget token"** action? Without it, disconnecting an MCP feels a bit bizarre. I also noticed it's explicitly mentioned as the third point in the **[RHIDP-14843](https://redhat.atlassian.net/browse/RHIDP-14843)** description.

2. Would it be possible to eliminate the UI flicker that's happening when toggling the MCP on/off?

3. I also noticed that when the toggle is turned **off**, the status shown in the panel doesn't change. That doesn't seem to match the behavior shown in the prototype.

Also, from a UX perspective, I think the **Save** button should become enabled whenever the user changes the toggle or token. Right now, the changes seem to take effect immediately, leaving **Cancel** as the only meaningful action, which was a bit confusing to me as a first-time user.

cc: @ShiranHi 

https://github.com/user-attachments/assets/08488804-8acc-471c-9807-67cb44abf131

**Comment by ciiay** [comment]
Hi @HusneShabbir , thanks for the review. Here's some background from previous discussion.


- Button Renaming: Rename the “Forget token” button to “Remove personal token,” as tokens defined in the app-config cannot be removed or forgotten.
- Button Visibility: Display the “Remove personal token” button only when a personal token has been provided by the user.
- Token Priority & Fallback: User-provided tokens override the default app-config token. If a user removes their personal token for a server that has a default app-config token, the system will revert to connecting using that preserved default token.


> 1. Why haven't we implemented the "forget token" action? Without it, disconnecting an MCP feels a bit bizarre. I also noticed it's explicitly mentioned as the third point in the [RHIDP-14843](https://redhat.atlassian.net/browse/RHIDP-14843) description.

It's implemented as "Remove personal token" and hidden for MCP servers that doesn't have a personal token input yet.

> 2. Would it be possible to eliminate the UI flicker that's happening when toggling the MCP on/off?
> 3. I also noticed that when the toggle is turned off, the status shown in the panel doesn't change. That doesn't seem to match the behavior shown in the prototype.

Good catch, fixed in my new commit.

> Also, from a UX perspective, I think the Save button should become enabled whenever the user changes the toggle or token. Right now, the changes seem to take effect immediately, leaving Cancel as the only meaningful action, which was a bit confusing to me as a first-time user.

Addressed in my new commit 🤝

**Review by HusneShabbir — COMMENTED** [review]
Tested all the edge cases locally.
/lgtm

FYI: I'll cover the E2E tests for this implementation in [RHIDP-15136.](https://redhat.atlassian.net/browse/RHIDP-15136)

**Review by HusneShabbir — COMMENTED** [review]
Everything looks good to me. Thanks for addressing my review feedback!

I do have one observation, though: shouldn't the status of the DCR-authenticated MCP server match exactly what is shown in the sidebar?

Why does it show **"Token required"** even when the server is connected?

Image

https://github.com/user-attachments/assets/fe9241b5-97a1-4576-92e2-d43c62a76d91

**Comment by ciiay** [comment]
Hi @HusneShabbir , thanks for the review.

>Why does it show "Token required" even when the server is connected?

This is the DCR auth flow that was not covered in my previous code changes. I have added fix for this flow in my latest commit. Can you take another look?

**Review by HusneShabbir — COMMENTED** [review]
/lgtm

**Review by HusneShabbir — COMMENTED** [review]
@ciiay I think it would be great if we could implement a similar UI for DCR-authenticated MCPs as well—showing the list of tools, MCP status, and a toggle.

The only difference for DCR would be the authentication messaging. Instead of asking for a token, we could display:

> **This server uses Dynamic Client Registration (DCR). Tokens are minted automatically using your Backstage identity—no manual token is required.**

**Review by HusneShabbir — COMMENTED** [review]
The E2E tests for https://redhat.atlassian.net/browse/RHIDP-15136 are included in https://github.com/ciiay/rhdh-plugins/pull/4. If we move forward with the DCR UI updates, the tests will only need a few minor changes, which I'll take care of.

I also reviewed https://github.com/ciiay/rhdh-plugins/pull/4 and the CI looks clean to me. Please feel free to merge it.

**Comment by ciiay** [comment]
Hi @HusneShabbir , I agree that we should align up the MCP server modal layout for both personal token authentication and dcr authentication. Here's some screenshots for that update.

image

image

image

**Review by HusneShabbir — APPROVED** [review]
/lgtm

Appreciate your patience and the effort in addressing all the PR comments. We're good to get this merged.

File: workspaces/intelligent-assistant/plugins/intelligent-assistant/src/components/McpServersSettings.tsx [hunk 1]
```diff
@@ -1026,128 +1539,136 @@ export const McpServersSettings = ({
...
-                    className={classes.forgetTokenButton}
-                  >
-                    {t('mcp.settings.removePersonalToken')}
-                  </Button>
+            )}
+            {showPersonalTokenField && (
+              <FormGroup
+                label={t('mcp.settings.authenticationToken')}
+                fieldId="mcp-pat-input"
+              >
```
Comment (karthikjeeyar) [inline workspaces/intelligent-assistant/plugins/intelligent-assistant/src/components/McpServersSettings.tsx]: `FormGroup` must be wrapped with `Form` element,  only then the spacing will be inherited from PF.

```suggestion
            <Form>
              <FormGroup
                label={t('mcp.settings.authenticationToken')}
                fieldId="mcp-pat-input"
              >
```




**Before: (with PR changes):** 

Image

Error state:

Image

---




**After: wrapping with `<Form>`:**


Image

Error state:

Image

Note that Form title gets bolder and PF Spacing is applied after wrapping the formgroup with form.

**Review by karthikjeeyar — CHANGES_REQUESTED** [review]
Looks great functionally 👏🏻  I have tested the behavior and flows are working well.




As this PR is mainly focussed on refactoring, I have a few suggestions: 

1. could we move the component to a more PatternFly-native structure?  Right now, the mix of custom div's/styles seems to be what causes some spacing inconsistencies.
    
    A few things that could help:
    
    - Use PF layout primitives (`Stack, StackItem, Flex, FlexItem`) for section structure instead of `div`.
    - Render the tools list with PF `List/ListItem` instead of native `ul/li`.
    - Wrap the token input content under PF `Form` element,  so spacing follows PF defaults.
    - Reduce custom modal spacing overrides and rely more on PF spacing tokens

2. `McpServersSettings` is doing API orchestration + modal state + rendering in one place. Consider extracting modal logic into a dedicated hook/component for readability and easier testing. 

This would keep the current functionality intact while improving consistency and maintainability.

**Comment by ciiay** [comment]
Hi @karthikjeeyar , thanks for the review. I have addressed your comments. PTAL, thanks 🤝

**Review by karthikjeeyar — APPROVED** [review]
Thanks @ciiay, I've verified locally.

/approve
/lgtm
